RecoMET/METProducers/src/HcalNoiseInfoProducer.cc

```diff
@@ -428,39 +593,45 @@ HcalNoiseInfoProducer::filldigis(edm::Event& iEvent, const edm::EventSetup& iSet
   // get total charge in calibration channels
   if(hCalib.isValid() == true)
   {
+
+
+     std::vector<std::vector<int> > lasmon_adcs(laserMonCBoxList_.size(), std::vector<int>());
+     std::vector<std::vector<int> > lasmon_capids(laserMonCBoxList_.size(), std::vector<int>());
+
      for(HcalCalibDigiCollection::const_iterator digi = hCalib->begin(); digi != hCalib->end(); digi++)
      {
         if(digi->id().hcalSubdet() == 0)
            continue;
 
-        /*
-        HcalCalibDetId cell = digi->id();
-        DetId detcell = (DetId)cell;
-        
-        const HcalChannelStatus* mydigistatus=myqual->getValues(detcell.rawId());
-
-        if(mySeverity->dropChannel(mydigistatus->getValue()))
-           continue;
-        if(digi->zsMarkAndPass())
-           continue;
-
-        const HcalQIECoder *channelCoder = conditions->getHcalCoder(cell);
-	const HcalQIEShape* shape = conditions->getHcalShape(channelCoder);
-        HcalCoderDb coder(*channelCoder, *shape);
-
-        CaloSamples tool;
-        coder.adc2fC(*digi, tool);
-
-        for(int i = 0; i < (int)digi->size(); i++)
-           TotalCalibCharge = TotalCalibCharge + tool[i];
-        */
-
-	// Original code computes total calib charge over all digis.  While I think it would be more useful to skip
-	// zs mark-and-pass channels, I keep this computation as is.  Individual HBHE and HF variables do skip
-	// the m-p channels.  -- Jeff Temple, 6 December 2012
-
-	for(int i = 0; i < (int)digi->size(); i++)
-	  TotalCalibCharge = TotalCalibCharge + adc2fC[digi->sample(i).adc()&0xff];
+        // Fill the lasermonitor channels
+        if( fillLaserMonitor_ ) {
+          int cboxch  = digi->id().cboxChannel( );
+          int iphi    = digi->id().iphi();
+          int ieta    = digi->id().ieta();
+         
+          // only check channels having the requested cboxch
+          if (std::find( laserMonCBoxList_.begin(), laserMonCBoxList_.end(),
```

Comment:
just to clarify the previous discussion - "find" is effectively looping over your container - so its not that you are just looking at the 4-8 channels of interest, your looking at all of them (and 4-8 of them, you are looking twice)

Reply:
Right, there are two cases
For the channels of interest the std::find should return true on the first iteration and then perform the inner loop
For all other channels the std::find makes one loop and returns false

so yes, its clear that the std::find does not help, but it does not hurt much.  I can remove it if you like